packages/devtools_app/lib/src/debugger/flutter/debugger_screen.dart

```diff
@@ -60,13 +73,128 @@ class DebuggerScreenBodyState extends State<DebuggerScreenBody> {
       axis: Axis.horizontal,
       initialFirstFraction: 0.25,
       // TODO(https://github.com/flutter/devtools/issues/1648): Debug panes.
-      firstChild: const Text('Debugger details'),
+      firstChild: ScriptPicker(
+        scripts: scriptList,
+        onSelected: onScriptSelected,
+        selected: loadingScript,
+      ),
       // TODO(https://github.com/flutter/devtools/issues/1648): Debug controls.
-      secondChild: CodeView(
-        script: script,
+      secondChild: loadingScript != null && script == null
+          ? const Center(child: CircularProgressIndicator())
+          : CodeView(
+              script: script,
+            ),
+    );
+  }
+}
+
+/// Picker that takes a [ScriptList] and allows selection of one of the scripts inside.
+class ScriptPicker extends StatefulWidget {
+  const ScriptPicker({Key key, this.scripts, this.onSelected, this.selected})
+      : super(key: key);
+
+  final ScriptList scripts;
+  final void Function(ScriptRef scriptRef) onSelected;
+  final ScriptRef selected;
+
+  @override
+  ScriptPickerState createState() => ScriptPickerState();
+}
+
+class ScriptPickerState extends State<ScriptPicker> {
+  List<ScriptRef> _filtered;
+  TextEditingController filterController = TextEditingController();
+
+  @override
+  void initState() {
+    super.initState();
+    if (_isNotLoaded) initFilter();
+  }
+
+  @override
+  void didUpdateWidget(ScriptPicker oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (_isNotLoaded) {
+      initFilter();
+    } else if (oldWidget.scripts != widget.scripts) {
+      updateFilter(filterController.text);
+    }
+  }
+
+  void initFilter() {
+    // Make an educated guess as to the main package to slim down the initial list of scripts we show.
+    if (widget.scripts?.scripts != null) {
+      final mainFile = widget.scripts.scripts
+          .firstWhere((ref) => ref.uri.contains('main.dart'));
+      filterController.text = mainFile.uri.split('/').first;
+      updateFilter(filterController.text);
+    }
+  }
+
+  void updateFilter(String value) {
+    setState(() {
+      if (widget.scripts?.scripts == null) {
+        _filtered = null;
+      } else {
+        _filtered = widget.scripts.scripts
+            .where(
+              (ref) => ref.uri.contains(value.toLowerCase()),
+            )
+            .toList();
+      }
+    });
+  }
+
+  Widget _buildScript(ScriptRef ref) {
+    final selectedColor = Theme.of(context).selectedRowColor;
+    return Material(
+      color: ref == widget.selected ? selectedColor : null,
+      child: InkWell(
+        onTap: () => widget.onSelected(ref),
+        child: Padding(
```

Comment:
nit: if you have `Padding` + `Align` why not use `Container`
@pq this is an example where I think we could have a nice quick fix to convert this code to a `Container` if we thought that was the cleaner way to write this.

Reply:
Interesting!

If we can generalize this at all, it would be nice to add some layout related refactorings to server.

/fyi @bwilkerson.

Reply:
That's a reasonable refactoring to apply here.

I don't think it's a particularly high-value lint or quick fix.  It's not a very common problem, and there can be cases where the transformation between Padding(child: Align)) to Container() is not possible.